tests/sentry/integrations/slack/test_utils.py

```diff
@@ -103,3 +103,193 @@ def test_rate_limiting(self):
         )
         with pytest.raises(ApiRateLimitedError):
             get_channel_id(self.organization, self.integration, "@user")
+
+
+class GetChannelIdFasterTest(TestCase):
+    @with_feature("organizations:slack-use-new-lookup")
+    def setUp(self):
+        self.resp = responses.mock
+        self.resp.__enter__()
+
+        self.integration = install_slack(self.event.project.organization)
+        self.add_list_response(
+            "conversations",
+            [
+                {"name": "my-channel", "id": "m-c"},
+                {"name": "other-chann", "id": "o-c"},
+                {"name": "my-private-channel", "id": "m-p-c", "is_private": True},
+            ],
+            result_name="channels",
+        )
+        self.add_list_response(
+            "users",
+            [
+                {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+                {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+                {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+            ],
+            result_name="members",
+        )
+
+    # self.add_msg_response("other-chann", "o-c")
+    # self.add_list_response(
+    #     "users",
+    #     [
+    #         {"name": "first-morty", "id": "m", "profile": {"display_name": "Morty"}},
+    #         {"name": "other-user", "id": "o-u", "profile": {"display_name": "Jimbob"}},
+    #         {"name": "better_morty", "id": "bm", "profile": {"display_name": "Morty"}},
+    #     ],
+    #     result_name="members",
+    # )
+
+    @with_feature("organizations:slack-use-new-lookup")
+    def tearDown(self):
+        self.resp.__exit__(None, None, None)
+
+    def add_list_response(self, list_type, channels, result_name="channels"):
+        self.resp.add(
+            method=responses.GET,
+            url="https://slack.com/api/%s.list" % list_type,
+            status=200,
+            content_type="application/json",
+            body=json.dumps({"ok": True, result_name: channels}),
+        )
+
+    def add_msg_response(self, channelname, channelid, result_name="channel"):
+        self.resp.add(
+            method=responses.POST,
+            url="https://slack.com/api/chat.scheduleMessage",
+            status=200,
+            content_type="application/json",
+            body=json.dumps(
+                {"ok": True, result_name: channelid, "scheduled_message_id": "Q1298393284"}
+            ),
+        )
+
+    def add_bad_msg_response(self):
```

Comment:
Since this is almost the same as `add_msg_response` you could combine them and pass in an arg to determine whether the response is good or bad.